### PR TITLE
Extract Translation Keys from JavaScript Files (#1713)

### PR DESCRIPTION
**Pull request by miguelvaara**
Note: @osma , this is mainly for a design review and the Design PR can still be improved if you find the design difficult to review based on this level of descriptions.

## Link to relevant issue(s), if any
#1713 

- Closes #

## Description of the changes in this PR
A separate PHP class (JsTranslationExtractor.php) was created. By running it automatically during the extracting and using its built-in regex expression, translation keys can be read straigth from JavaScript files and Vue components (but naturally not from Vue templates).

The functionality requires creating getters for translation keys in the computed section, as shown below:

```
computed: {
  testTitle () {
    return $t('Test title in js file')
  },
  termCountsTitle () {
    return $t('Term counts by language 2')
  },
}
```

This approach is more structured (Managing translations in computed centralizes key handling and makes it possible to process them dynamically as variables if necessary)

In practice, if this works, we won’t need to add a vue-i18n dependency which will certainly simplify future maintenance.

Intended Workflow:
1) Read and extract new keys from Vue components.
2) Extraction writees to the same translation files that Vue components are already reading (the simples possible implementation).

So far, the setup has been run in "dry mode" and has not yet been tested for writing to the messages files:
```
php bin/console translation:extract en --prefix="" --dump-messages
```
The keys from JavaScript test file:
```
php bin/console translation:extract en --prefix="" --dump-messages | grep 'Term counts by language'
Found key: Term counts by language 2
 * Term counts by language 2
 * Term counts by language
```

## Known problems or uncertainties in this PR
The directory containing the test JavaScript file is still in the wrong location, so additional work is required to fix the paths.

## Checklist

- [ ] phpUnit tests pass locally with my changes
- [ ] I have added tests that show that the new code works, or tests are not relevant for this PR (e.g. only HTML/CSS changes)
- [ ] The PR doesn't reduce accessibility of the front-end code (e.g. tab focus, scaling to different resolutions, use of `.sr-only` class, color contrast)
- [ ] The PR doesn't introduce unintended code changes (e.g. empty lines or useless reindentation)

### CODE REVIEW

**Review by miguelvaara — COMMENTED** [review]
@osma 

The code has now been brought closer to the quality required for design review before the final code changes.

**Please pay attention to the following:**

**Remember to run:**
- `npm install`
- `php composer.phar update`
- `php bin/console cache:clear`

**Comments:**
- The location and naming of the class that enables the extraction of keys from JavaScript files (src/Translation/Extractor/JsTranslationExtractor.php).
- To facilitate testing, additional placeholder texts "(test case [A-E])" have been added to some translation keys. Modify the letter identifiers as needed and run: ```php bin/console translation:extract en --prefix="" --dump-messages -vvv```
- So far the messages files have not been updated. Only "dry runs" have been performed, but the "real runs" are likely the next step to ensure proper functionality.
- The JavaScript files directory is provided as an argument in the _services.yaml_ file, so hardcoding at the code level was avoided.
- A logical next step for the following issue would be configuring the push to Lokalise. Does the work done here support that goal?

**Review by miguelvaara — COMMENTED** [review]
**Review Instructions**

**Copy and run the following:**

```
git fetch
git switch issue1713-what-is-the-best-way-to-extract-translation-keys-from-vue-components
git pull
php composer.phar update
npm install
```

**Translation key extraction from Vue components**

You can test the functionality by modifying a translation key in the file _resource/js/term-counts.js_. For example, change the key _return $t('Term counts by language')_. Then run the command `php bin/console translation:extract en --prefix="" --force --format=json` in the root of the Skosmos project. This will add the modified translation key to the file _resource/translations/messages.en.json_.

**Additional Note:**

The tests _SonarCloud Code Analysis_ and _codecov/project_ are failing. How do you approach this?

File: src/Translation/Extractor/JsTranslationExtractor.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    private string $prefix = '';
+
+    public function setPrefix(string $prefix): void
+    {
+        $this->prefix = $prefix ?: '';
+    }
+
+    public function extract($directory, MessageCatalogue $catalogue): void
+    {
+        $directory = $this->directory;
```
Comment (osma) [inline src/Translation/Extractor/JsTranslationExtractor.php:24]: SonarCloud [complains](https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=1716&id=NatLibFi_Skosmos&open=AZO2xxKtw_CSfO7Da1EM) about overriding the $directory parameter.

I don't think there's any need to assign `$directory = $this->directory;`, the code below could just directly access `$this->directory`, avoiding the problem.

**Review by osma — CHANGES_REQUESTED** [review]
It works as it should, but I think the code needs some further cleanup.
Also, I suggested renaming the file to better fit in with Skosmos file naming conventions.

File: src/Translation/Extractor/JsTranslationExtractor.php (hunk 1 above)
Comment (miguelvaara) [reply src/Translation/Extractor/JsTranslationExtractor.php:24]: _$directory = $this->directory;_ removed and it works.

File: src/Translation/Extractor/JsTranslationExtractor.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        echo "Using directory: $directory" . PHP_EOL;
```
[the remark this answers is not in the document]
Comment (miguelvaara) [reply src/Translation/Extractor/JsTranslationExtractor.php]: Fixed

File: src/Translation/Extractor/JsTranslationExtractor.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            return;
+        }
+        $iterator = new \RecursiveIteratorIterator(
+            new \RecursiveDirectoryIterator($directory)
+        );
+        foreach ($iterator as $file) {
+            if ($file->getExtension() !== 'js') {
+                continue;
+            }
+            echo "Processing file: " . $file->getPathname() . PHP_EOL;
```
[the remark this answers is not in the document]
Comment (miguelvaara) [reply src/Translation/Extractor/JsTranslationExtractor.php]: Fixed

[the remark this answers is not in the document]
Comment (miguelvaara) [reply src/Translation/Extractor/JsTranslationExtractor.php]: Fixed

**Review by osma — APPROVED** [review]
LGTM

**Pull request merged**